concept/validation/schema-validation.feature

```diff
@@ -11,490 +11,734 @@ Feature: Schema validation
+    When relation(rel0c) unset annotation: @abstract
+    Then relation(rel0c) get roles contain:
+      | relation:role |
+    Then relation(rel0c) get declared roles is empty
+    Then transaction commits; fails
 
 
-  Scenario: Concrete relation types must relate at least one role, but these may be inherited
+  Scenario: Concrete relation types must relate at least one role
     Given create relation type: rel00
-    Given relation(rel00) set abstract: true
-    Given relation(rel00) set relates role: role00
+    Given relation(rel00) set annotation: @abstract
+    Given relation(rel00) create role: role00
     Given create relation type: rel01
-    Given relation(rel01) set abstract: true
+    Given relation(rel01) set annotation: @abstract
     Given transaction commits
 
-    When session opens transaction of type: write
+    When connection open schema transaction for database: typedb
     When create relation type: rel1
     When relation(rel1) set supertype: rel01
-    Then transaction commits; throws exception
+    Then relation(rel1) get roles contain:
+      | relation:role |
+    Then relation(rel1) get declared roles is empty
+    Then transaction commits; fails
 
-    When session opens transaction of type: write
+    When connection open schema transaction for database: typedb
     When create relation type: rel1
+    When relation(rel1) set supertype: rel01
+    Then relation(rel1) get roles contain:
+      | relation:role |
+    Then relation(rel1) get declared roles is empty
     When relation(rel1) set supertype: rel00
-    Then transaction commits
-
-    When session opens transaction of type: write
-    When relation(rel00) unset related role: role00
-    Then transaction commits; throws exception
-
-    When session opens transaction of type: write
+    Then relation(rel1) get roles contain:
+      | rel00:role00 |
+    Then relation(rel1) get declared roles is empty
+    When transaction commits
+
+    When connection open schema transaction for database: typedb
+    When relation(rel00) delete role: role00
+    Then relation(rel00) get roles contain:
+      | relation:role |
+    Then relation(rel1) get roles contain:
+      | relation:role |
+    Then relation(rel00) get roles do not contain:
+      | rel00:role00 |
+    Then relation(rel1) get roles do not contain:
+      | rel00:role00 |
+    Then transaction commits; fails
+
+    When connection open schema transaction for database: typedb
     When relation(rel1) set supertype: rel01
-    Then transaction commits; throws exception
+    Then relation(rel1) get roles contain:
+      | relation:role |
+    Then relation(rel1) get roles do not contain:
+      | rel00:role00 |
+    Then transaction commits; fails
 
 
-  Scenario: Relation types may not declare a role with the same name as one declared by its ancestors
+  Scenario: Relation types may not declare a role with the same name as one declared in its inheritance line
```

Comment:
Should be probably covered in a simple case in relationtype.feature, but can substitute it by a cooler test like this after the move.